### DISCUSSION TITLE
Cannot seed with `updateOrCreate()` a sluggable model with a slug

### DISCUSSION THREAD

**Discussion by tchapi** [area17/twill#1004] [Q&A]
Hello

I'm trying to seed my database by using Twill's ModuleRepository methods.

For instance, I can do this :

    use A17\Twill\Repositories\SettingRepository; 

        $settingRepository->updateOrCreate([
            'key' => 'google_analytics_id',
            'section' => 'general',
        ], [
            'key' => 'google_analytics_id',
            'section' => 'general',
            'en' => [
                'active' => true,
                'locale' => 'en',
                'value' => 'UA-TEST',
            ],
        ]);

But if I try to seed a sluggable model, it fails : 

    use App\Repositories\CategoryRepository;

    [...]

        $categoryRepository->updateOrCreate([
            'title' => 'Events',
        ], [
            'title' => 'Events',
            'published' => true,
            'slug' => [
                 'en' => 'events',
             ],
        ]);

with : `Column not found: 1054 Unknown column 'slug' in 'field list' ...` which is pretty normal since the column is in another table.

It looks like the "slug" key should be in the `getReservedFields()` function of the `ModuleRepository`, no?

I think I'm using the correct array structure since this is what the `afterSaveHandleSlugs($object, $fields)` function of the `HandleSlugs` trait expect to have to create or update the slug afterwards.

Thanks for your help !
BR

**Comment by ifox** [comment]
Hi @tchapi,

You are using the correct structure.

When creating a model, only fillable fields should be sent through the underlying SQL query, so I suspect that you might have `slug` in your `Category` model fillable array.

**Comment by tchapi** [comment]
Well I checked and I have not : 

```
<?php

namespace App\Models;

use A17\Twill\Models\Behaviors\HasSlug;
use A17\Twill\Models\Model;

class Category extends Model
{
    use HasSlug;

    protected $fillable = [
        'published',
        'title',
    ];

    public $slugAttributes = ['title'];
    public $checkboxes = ['published'];
}
```
What other thing might I have overlooked ?

**Comment by tchapi** [comment]
Hi, any hints for me regarding this issue by any chance ?
Thanks !

**Comment by yanhao-li** [comment]
Hi @tchapi ,

As @ifox mentioned, `ModuleRepository` is using `fill` to do the mass assignment: https://github.com/area17/twill/blob/1.2/src/Repositories/ModuleRepository.php#L186

However, since you haven't put `slug` in your `$fillable`, this issue is not expected and weird.
I have just created a fresh Twill project followed exactly your model setting, and tried your seed operation in tinker, didn't encounter any issues.

I would recommend you try using `->toSql()` to get the raw SQL query to find where goes wrong, or you could post your entire project that I could help to take a look as well.

Hope this helps!

**Comment by tchapi** [comment]
Hello @yanhao-li and thanks for your answer. I will try `toSql()` and see how it goes.

**Comment by tchapi** [comment]
Ok I think I found the culprit : https://github.com/laravel/framework/blob/5.8/src/Illuminate/Database/Console/Seeds/SeedCommand.php#L62-L64

When running the `database:seed` command, the Model is unguarded on purpose, which means `isFillable()` returns `true` for all attributes, and ignores the `fillable` array, just like a `forceFill()`.

I've checked by `var_dump`ing everywhere and that is what happens in my case: `Model::$unguarded` is true when seeding.

Am I missing something ?
Thanks again !
